### ISSUE TITLE
Can't config destinationFolder

### ISSUE THREAD

**Issue by ColdSiemens** [mancku/strapi-plugin-schemas-to-ts#54]
I can't configure `destinationFolder` property

Strapi: 4.21.0
strapi-plugin-schemas-to-ts: 1.3.2

`plugins.js`:
```
module.exports = () => ({
  'schemas-to-ts': {
    enabled: true,
    config: {
      destinationFolder: "generated-interfaces",
    }
  },
});
```

`Error: schemas-to-ts ⚠️  The given destinationFolder is inside the Strapi dist at DestinationPaths.assertDestinationIsNotStrapiCoreFolder`

<img width="742" alt="image" src="https://github.com/mancku/strapi-plugin-schemas-to-ts/assets/29114556/56648165-52cb-404a-bef6-3b508e4bea4e">

**Comment by mancku** [comment]
Hi @ColdSiemens I've run some tests with that same version of Strapi, version of the plugin and plugin configuration, and it worked, so I think I need a little more information to be able to reproduce your problem.

<details>
  <summary>This is on Windows:</summary>

> DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\api\about\content-types\about\schema.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\api\category\content-types\category\schema.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\api\customer\content-types\customer\schema.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\api\home\content-types\home\schema.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\api\order\content-types\order\schema.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\api\privacy-policy\content-types\privacy-policy\schema.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\api\review\content-types\review\schema.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\api\user-info\content-types\user-info\schema.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\api\vehicle\content-types\vehicle\schema.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\api\warranty\content-types\warranty\schema.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\components\seo-components\meta.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\components\ui-components\hero.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\components\vehicles-e-commerce\address.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\components\vehicles-e-commerce\engines-specifications.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\components\vehicles-e-commerce\payment-information.json
DEBUG: Converting schema C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\src\components\vehicles-e-commerce\vehicle-features.json
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\api\about.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\api\category.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\api\customer.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\api\home.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\api\order.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\api\privacy-policy.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\api\review.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\api\user-info.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\api\vehicle.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\api\warranty.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\components\seo-components\Meta.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\components\ui-components\Hero.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\components\vehicles-e-commerce\Address.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\components\vehicles-e-commerce\EnginesSpecifications.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\components\vehicles-e-commerce\PaymentInformation.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\components\vehicles-e-commerce\VehicleFeatures.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\common\Payload.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\common\User.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\common\MediaFormat.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\common\Media.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\common\AdminPanelRelationPropertyModification.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\common\BeforeRunEvent.ts
DEBUG: Writing file C:\Dev\mancku\test-strapi-plugin-schemas-to-ts\generated-interfaces\common\AfterRunEvent.ts
</details>

<details>
  <summary>This is on Linux:</summary>

> DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/api/about/content-types/about/schema.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/api/category/content-types/category/schema.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/api/customer/content-types/customer/schema.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/api/home/content-types/home/schema.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/api/order/content-types/order/schema.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/api/privacy-policy/content-types/privacy-policy/schema.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/api/review/content-types/review/schema.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/api/user-info/content-types/user-info/schema.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/api/vehicle/content-types/vehicle/schema.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/api/warranty/content-types/warranty/schema.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/components/seo-components/meta.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/components/ui-components/hero.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/components/vehicles-e-commerce/address.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/components/vehicles-e-commerce/engines-specifications.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/components/vehicles-e-commerce/payment-information.json
DEBUG: Converting schema /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/src/components/vehicles-e-commerce/vehicle-features.json
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/api/about.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/api/category.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/api/customer.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/api/home.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/api/order.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/api/privacy-policy.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/api/review.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/api/user-info.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/api/vehicle.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/api/warranty.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/components/seo-components/Meta.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/components/ui-components/Hero.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/components/vehicles-e-commerce/Address.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/components/vehicles-e-commerce/EnginesSpecifications.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/components/vehicles-e-commerce/PaymentInformation.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/components/vehicles-e-commerce/VehicleFeatures.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/common/Payload.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/common/User.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/common/MediaFormat.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/common/Media.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/common/AdminPanelRelationPropertyModification.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/common/BeforeRunEvent.ts
DEBUG: Writing file /home/mancku/dev/mancku/test-strapi-plugin-schemas-to-ts/generated-interfaces/common/AfterRunEvent.ts
</details>

**Comment by andreasrippl** [comment]
Hi I've experienced the same issue any path given inside the strapi project that is used as a destinationFolder keeps giving the same "The given destinationFolder is inside the Strapi dist" because it actually also is.

I tried to dig in and the strapiPaths.dist.root in the destinationPaths.js is my **root path of my project directory**. So in return every path i can give the destinationPath (as it is not allowed to give it a path outside the project) starts with this root path.

I am not sure if i can just set the dist path to be something different other than the root tho (couldn't see an obvious way somwhere in the docs) because as stated here: https://docs.strapi.io/dev-docs/configurations/typescript#project-structure-and-typescript-specific-configuration-files **distDir = root** if i am not reading this wrong. 

So not sure what the procedure is but i think this rule 
https://github.com/mancku/strapi-plugin-schemas-to-ts/blob/2e92a1616fb1137bec4912b346a35fed3fd477d9/server/models/destinationPaths.ts#L96-L98 
Would need to be removed or changed to not make it fail in the default setup if a destinationPath is added. If removed everything seems to work fine. 

To reproduce: 
1. Quickstart new strapi project
2. Install the extension 
3. Set some destinationPath thats within the project

**Comment by Bredansky** [comment]
I've faced the same issue with the same versions of packages on macOS 14.3.1 

`plugins.js`:
```
export default {
    'schemas-to-ts': {
        enabled: true,
        config: {
          destinationFolder: "generated-interfaces",
        }
      },
  }
```

`destinationPaths.js`:
```
        if (destinationFolder.startsWith(strapiPaths.dist.root)) {
            console.log(destinationFolder);
            console.log(strapiPaths.dist.root)
            throw new Error(`${pluginName_1.pluginName} ⚠️  The given destinationFolder is inside the Strapi dist`);
        }
 ```
 
 Output:
 ```
/cms/generated-interfaces
/cms
[ERROR]  There seems to be an unexpected error, try again with --debug for more information 
Error: schemas-to-ts ⚠️  The given destinationFolder is inside the Strapi dist
```

**Comment by ColdSiemens** [comment]
My end goal was to create a mono-repo with Strapi and Next.js with TypeScript.
Since I had an issue with this plugin and it hasn't satisfied my needs, I've used a [workaround](https://strapi.io/blog/improve-your-frontend-experience-with-strapi-types-and-type-script) to generate types and added my own thing to it to constantly update types on my front-end part of the repo when Strapi is in the development mode or once after the build:

After finishing [tutorial](https://strapi.io/blog/improve-your-frontend-experience-with-strapi-types-and-type-script) part let's add a new script to the `backend/package.json` and update a `build` script with it:

```json
"updateTypes": "yarn strapi ts:generate-types && cd .. && yarn run copytypes"
"build": "strapi build && yarn updateTypes",
```

Next update `module.exports` in `backend/src/bootstrap.js` to be smth like this:

```js
module.exports = async () => {
  const shouldImportSeedData = await isFirstRun();
  if (shouldImportSeedData) {
    try {
      await importSeedData();
    } catch (error) {
      console.log("Could not import seed data");
      console.error(error);
    }
  }
  exec('yarn updateTypes', (error, stdout, stderr) => {
    if (error) {
      console.log(`error: ${error.message}`);
      return;
    }
    if (stderr) {
      console.log(`stderr: ${stderr}`);
      return;
    }
    console.log(`stdout: ${stdout}`);
  });
};
```

Basically, I'm calling a script to trigger types changes in my whole project.

Also, I've added these lines to `.gitignore` in the project root folder since there's no need to keep them around:

```
**/components.d.ts
**/contentTypes.d.ts
```

As a result, I now have a working TS connection between Next and Strapi, which ensures that my front-end fields are always correct.

I know it could be improved but whatsoever I hope it can be helpful for somebody

**Comment by Willen17** [comment]
> My end goal was to create a mono-repo with Strapi and Next.js with TypeScript. Since I had an issue with this plugin and it hasn't satisfied my needs, I've used a [workaround](https://strapi.io/blog/improve-your-frontend-experience-with-strapi-types-and-type-script) to generate types and added my own thing to it to constantly update types on my front-end part of the repo when Strapi is in the development mode or once after the build:
> 
> After finishing [tutorial](https://strapi.io/blog/improve-your-frontend-experience-with-strapi-types-and-type-script) part let's add a new script to the `backend/package.json` and update a `build` script with it:
> 
> ```json
> "updateTypes": "yarn strapi ts:generate-types && cd .. && yarn run copytypes"
> "build": "strapi build && yarn updateTypes",
> ```
> 
> Next update `module.exports` in `backend/src/bootstrap.js` to be smth like this:
> 
> ```js
> module.exports = async () => {
>   const shouldImportSeedData = await isFirstRun();
>   if (shouldImportSeedData) {
>     try {
>       await importSeedData();
>     } catch (error) {
>       console.log("Could not import seed data");
>       console.error(error);
>     }
>   }
>   exec('yarn updateTypes', (error, stdout, stderr) => {
>     if (error) {
>       console.log(`error: ${error.message}`);
>       return;
>     }
>     if (stderr) {
>       console.log(`stderr: ${stderr}`);
>       return;
>     }
>     console.log(`stdout: ${stdout}`);
>   });
> };
> ```
> 
> Basically, I'm calling a script to trigger types changes in my whole project.
> 
> Also, I've added these lines to `.gitignore` in the project root folder since there's no need to keep them around:
> 
> ```
> **/components.d.ts
> **/contentTypes.d.ts
> ```
> 
> As a result, I now have a working TS connection between Next and Strapi, which ensures that my front-end fields are always correct.
> 
> I know it could be improved but whatsoever I hope it can be helpful for somebody

Was looking for something like this. Great solution dude, thanks for sharing!
